src/main.rs

```diff
@@ -56,27 +51,50 @@ fn repl(calc: &mut Calc) {
                 if l.trim().is_empty() {
                     continue;
                 }
-                print_result(eval_line(l, calc));
+                if let Some(value) = eval_statement(l, calc) {
+                    println!("{}", value);
+                }
             }
             _ => break,
         }
     }
 }
 
-fn eval_line(input: &str, calc: &mut Calc) -> Result<Option<u64>, InterpError> {
-    debug!("input: {:?}", &input);
-    let ast = calc.from_str(input);
-    match ast {
-        Ok(ast_node) => {
-            debug!("AST: {:?}", &ast_node);
-            let bytecode = &mut vec![];
-            calc.to_bytecode(ast_node, bytecode);
-            debug!("Bytecode: {:?}", &bytecode);
-            match calc.eval(bytecode) {
-                Ok(result) => Ok(result),
-                Err(err) => Err(err),
+fn eval_statement(input: &str, calc: &mut Calc) -> Option<u64> {
+    let statements: Vec<String> = input
+        .replace("\n", "")
+        .split(";")
+        .map(|x| format!("{};", x))
+        .collect();
+
+    let mut result: Option<u64> = None;
```

Comment:
Is this type annotation required?

Reply:
removed :point_right: f105daa52900d10c8cf84f91e2b5c4707d3d022e